Python/sysmodule.c

```diff
@@ -924,15 +924,19 @@ static PyObject *
 call_trampoline(PyThreadState *tstate, PyObject* callback,
                 PyFrameObject *frame, int what, PyObject *arg)
 {
-    if (PyFrame_FastToLocalsWithError(frame) < 0) {
-        return NULL;
-    }
 
     PyObject *stack[3];
     stack[0] = (PyObject *)frame;
     stack[1] = whatstrings[what];
     stack[2] = (arg != NULL) ? arg : Py_None;
 
+    /* Discard any previous modifications the frame's fast locals */
+    if (frame->f_fast_as_locals) {
```

Comment:
Would it be safe to move this test inside PyFrame_FastToLocalsWithError()? Do nothing if it's zero?

Reply:
No. 
If we did that then `PyFrame_FastToLocalsWithError` would never do anything as `frame->f_fast_as_locals` would always be zero.